src/providers/eth_provider/state.rs

```diff
@@ -66,7 +66,10 @@ where
         // Convert the optional Ethereum block ID to a Starknet block ID.
         let starknet_block_id = self.to_starknet_block_id(block_id).await?;
         // Get the balance of the address at the given block ID.
-        self.starknet_provider().balance_at(starknet_address(address), starknet_block_id).await.map_err(Into::into)
+        self.starknet_provider()
+            .balance_at_native(starknet_address(address), starknet_block_id)
```

Comment:
I think you're replacing it wrongly
This should still be ETH here.

There is a difference between relayers (workers that relay EVM transactions) -> they pay in STRK and their balances are STRK

versus:

Kakarot users -> their native token remains ETH

Reply:
Got it, I understand.